### ISSUE TITLE
Compatibility

### ISSUE THREAD

**Issue by treanus** [DevelopmentalImagingMCRI/karawun#11]
Hi,

thank you for making available karawun.

I would like to know whether is works with iplan view/cranial version 3.0 and/or with Philips dicom data as template?

Thx,
Best regards,

Stefan.

**Comment by richardbeare** [comment]
Hi,
Thanks for your interest. We are using the same version of iplan, so can confirm that part works. We have a Siemens scanner and have not tested Philips dicom, however I do not anticipate any problems. The dicom template is used as a source of information that isn't available in the nifti files, and karawun does that by copying tags across. This ought to be quite vendor independent, provided there's nothing in private tags that your installation of brainlab looks for.

**Comment by treanus** [comment]
Hi,

I was able to create the dicoms. 

Important seems to be to convert the Philips multiframe enhanced dicom to a single frame dicom and only provide one slice to importTractgraphy as the dicom_template.

I still am not able to import the tracts in Iplan. The converted nii files get imported well. The tracts not. The iplan log-file reads: "Ignoring Dicom image series 'x' with modality 'SEG'", which is of course the problem. Did you change anything to iplan?

Thx,
Best regards,

Stefan.

**Comment by richardbeare** [comment]
That sounds like progress. We don't import direct to iPlan, but use the the main brainlab interface that provides patient selection icons as well as the choice of different brainlab tools, including iPlan and the image viewer. As a test, are you able to see the tracts in the image viewer? The viewer ought to be able to display the tracts in 3D, but only overlaying the reference image. You need to accept registration via the image fusion module in order to overlay on arbitary images. Also, are you able to show your conversion commandline? It sounds as though it is fine, but worth checking to make sure.

We might need to generate some more screenshots to guide the process.

**Comment by treanus** [comment]
Hi Richard,

thank you for the quick response!

The conversion command look as follows:
`importTractography -d Anonymized\ -\ 75320796/Rad\ Mr\ Taxie\ 01/Single\ -\ 8269/IM-0001-0001-0001.dcm -o iplan2 -n anat.nii.gz fa_reg2T1w.nii.gz -t DRT_L_select250000_incRN.tck DRT_R_select250000_incRN.tck -v`

with:

- "Anonymized\ -\ 75320796/Rad\ Mr\ Taxie\ 01/Single\ -\ 8269/IM-0001-0001-0001.dcm" is a single slice derived from a scout of the subject (converted from enhanced Philips multi-frame dicom)
- "anat.nii.gz" is my T1w image (skull stripped)
- fa_reg2T1w.nii.gz is an FA with dimension and voxel sizes of my diffusion dataset, but rigidly registered to the T1w (the whole DTI data is registered to the T1w, before doing tckgen)
- I have two tracts (DentatoRubroThalamic tracts)

Then, I use the following brainlab tools iPlan View 3.0 and/or iPlan Cranial 3.0 (see screenshot):

![Schermafbeelding 2020-07-31 om 15 19 01](https://user-images.githubusercontent.com/36293632/89100253-ea535700-d3f5-11ea-9fb0-f1e2ee37e882.png)
The iplan 2 output is copied the f:\PatientData\mrtrix folder.

PatientName and ID are recognised:
![Schermafbeelding 2020-07-31 om 15 19 23](https://user-images.githubusercontent.com/36293632/89100310-5766ec80-d3f6-11ea-9cd3-520f4cf623d9.png)

The tracts do not load. The log-file shows:
![Schermafbeelding 2020-08-01 om 12 42 33](https://user-images.githubusercontent.com/36293632/89100314-63eb4500-d3f6-11ea-81bf-cf61ebdd1320.png)


I could share my data with you, if that helps.

Thx in advance,

Stefan.

**Comment by treanus** [comment]
Just another point: I had to rescale the anatomical image.

My original T1 is a float 32, which a large intensity range. See screenshot.

![Schermafbeelding 2020-08-01 om 13 02 46](https://user-images.githubusercontent.com/36293632/89100628-ea088b00-d3f8-11ea-81e5-79f8dfc92ef0.png)


If I use that one, import fails:
```
Traceback (most recent call last):
  File "/Users/xm52195/anaconda3/envs/KarawunEnv/lib/python3.6/site-packages/pydicom/tag.py", line 30, in tag_in_exception
    yield
  File "/Users/xm52195/anaconda3/envs/KarawunEnv/lib/python3.6/site-packages/pydicom/filewriter.py", line 541, in write_dataset
    write_data_element(fp, dataset.get_item(tag), dataset_encoding)
  File "/Users/xm52195/anaconda3/envs/KarawunEnv/lib/python3.6/site-packages/pydicom/filewriter.py", line 483, in write_data_element
    writer_function(buffer, data_element, writer_param)
  File "/Users/xm52195/anaconda3/envs/KarawunEnv/lib/python3.6/site-packages/pydicom/filewriter.py", line 232, in write_numbers
    "{0}\nfor data_element:\n{1}".format(str(e), str(data_element)))
OSError: ushort format requires 0 <= number <= (32767 *2 +1)
```

I just rescaled using the command:

`mrcalc T1w_BrainExtractionBrain.nii.gz 100 -div anat.nii.gz`

Than importTractography works fine.

**Comment by richardbeare** [comment]
Sensible autoscaling of scans was something I ended up avoiding, but there probably should be some more checking. I did try using floats, but that is not part of the dicom standard. I also looked at slice dependent scaling, but that isn't either.

We don't import via iplan. We enter via "origin server", at least that is what it is called in our install:
![origin_serverA](https://user-images.githubusercontent.com/780449/89112360-019d4d80-d4a5-11ea-928a-60ec25195a43.png)

Then we go into "patient selection" and there's a manage dropdown on the right with an upload option:
![origin_serverB](https://user-images.githubusercontent.com/780449/89112373-22fe3980-d4a5-11ea-8f85-aef5c6165d96.png)

We upload our exported dicoms from the local machine to the server using that option, and we can use the karawun output in the previewing and image fusion tools before going into iplan cranial.

Does your installation look the same?

I am happy to try out your data if you still have problems.
